### PR TITLE
Feature etb#241 sun sensor

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: Drivers/seds/SunSensor.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <DeclaredInterfaceSet>
+      <Interface level="functional" name="SunSensorInterfaceType">
+        <CommandSet>
+          <Command name="InitSunSensorCmd" mode="sync">
+            <LongDescription>Initializes afec for reading Sun sensor data</LongDescription>
+            <Argument name="afecConfig" type="com/n7space/AfecHwas/AfecHwasConfig" mode="in" />
+          </Command>
+          <Command name="RequestDataCmd" mode="sync">
+            <LongDescription>Reads Sun sensor data</LongDescription>
+            <Argument name="dummy" type="com/n7space/Hwas/Word" mode="in" />
```
Comment (Lurkerpas) [inline Drivers/seds/SunSensor.xml]: Is this needed?

File: Drivers/seds/SunSensor.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<PackageFile xmlns="http://www.ccsds.org/schema/sois/seds" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.ccsds.org/schema/sois/seds seds.xsd">
+  <Package name="com/n7space/SunSensor">
+    <DeclaredInterfaceSet>
+      <Interface level="functional" name="SunSensorInterfaceType">
+        <CommandSet>
+          <Command name="InitSunSensorCmd" mode="sync">
+            <LongDescription>Initializes afec for reading Sun sensor data</LongDescription>
+            <Argument name="afecConfig" type="com/n7space/AfecHwas/AfecHwasConfig" mode="in" />
```
Comment (Lurkerpas) [inline Drivers/seds/SunSensor.xml]: What is configured here? Pin is fixed to 8, and statup + prescaler is overwritten during init.

File: Drivers/seds/SunSensorComponent.xml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                  <ArgumentValue name="analogChannel">
+                    <Value value="8" />
+                  </ArgumentValue>
+                </SendCommandPrimitive>
+              </Body>
+            </Activity>
+            <Activity name="ReadData">
+              <Argument name="chOutput" type="com/n7space/AfecHwas/ConversionData"></Argument>
+              <Body>
+                <SendCommandPrimitive command="ReturnDataCmd" interface="SunSensorReturn" shortDescription="Sun sensor is connected to PA19. It is AFE0_AD8. On Xplained Ultra board it is port J502 pin 3">
```
Comment (Lurkerpas) [inline Drivers/seds/SunSensorComponent.xml:72]: Can we do at least some "demo" processing... like normalization to 0-100 range or something. It does not need to be calibrated (although, use of Calibration would be cool).

File: Drivers/seds/SunSensor.xml (hunk 2 above)
Comment (jrachucki-n7s) [reply Drivers/seds/SunSensor.xml]: For simplicity, in the beginning, I decided to hardcode it and it seems like I forgot to change it back.

File: Drivers/seds/SunSensorComponent.xml (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/SunSensorComponent.xml:72]: Sure, I can do that.

File: Drivers/seds/SunSensor.xml (hunk 2 above)
Comment (Lurkerpas) [reply Drivers/seds/SunSensor.xml]: Hardcoding the config for this "virtual" sensor is OK.

Comment (jrachucki-n7s) [reply Drivers/seds/SunSensor.xml]: I have changed and tested it in new commit.

File: Drivers/seds/SunSensorComponent.xml (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/SunSensorComponent.xml:72]: Done. The AFEC has a 12-bit resolution. Thus the maximum value is 4096. To conduct normalization the output value is multiplied by 100 and divided by 4096

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**